lib/core/src/display/navigation/navigator.rs

```diff
@@ -90,10 +104,22 @@ impl Navigator {
             let y = pan.movement.y * scale;
             let z = 0.0;
 
-            properties_clone.mod_spring(|spring| { spring.fixed_point += Vector3::new(x, y, z); });
-        };
-
-        let dom_clone = dom.clone();
+            properties.mod_spring(|spring| { spring.fixed_point += Vector3::new(x, y, z); });
+        });
+
+        let mut dom_clone = dom.clone();
+        dom_clone.add_resize_callback(
+            enclose!((camera,mut properties) move |dimensions:&Vector2<f32>| {
+                camera.set_screen(dimensions.x, dimensions.y);
+                camera.update();
```

Comment:
Camera `set_screen` and update is already handled by the scene. Scene is the owner of the attached camera. This is not resposnibility of Navigator!

Reply:
@wdanilo yes, the logic is duplicated. But when this callback is ran, camera screen isn't updated, we need to be sure to update it.

Reply:
Then there is something wrong with responsibility. We should never duplicate such logic, because tracking what is happening is really hard. Camera is updated by scene. Scene tracks resize events of the dom. So maybe scene should emit onresize events instead and you should listen to them instead fo directly on dom ?